src/components/ChangeThumbnailButton.tsx

```diff
@@ -0,0 +1,257 @@
+  newThumbnailContent: Content | undefined;
+  newThumbnailIsReady: () => void;
+  pollNewThumbnail: (status: boolean) => void;
+}
+
+const StyledWrapper = styled.div`
+  display: flex;
+  margin-top: 1rem;
+`;
+
+const StyledCircularProgress = styled(CircularProgress)`
+  align-self: center;
+  margin-left: 1rem;
+`;
+
+const ResourceThumbnailUploaded = 'RESOURCE_THUMBNAIL_UPLOADED';
+
+const ChangeThumbnailButton: FC<ChangeThumbnailButtonProps> = ({
+  newThumbnailContent,
+  thumbnailUppy,
+  newThumbnailIsReady,
+  pollNewThumbnail,
+}) => {
+  const { t } = useTranslation();
+  const { values } = useFormikContext<Resource>();
+  const [fileInputIsBusy, setFileInputIsBusy] = useState(false);
+  const [showThumbnailDashboardModal, setShowThumbnailDashboardModal] = useState(false);
+  const [showPopover, setShowPopover] = useState(false);
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(null);
+  const [thumbnailUpdateError, setThumbnailUpdateError] = useState(false);
+
+  useEffect(() => {
+    if (thumbnailUppy) {
+      thumbnailUppy.on('upload-success', () => {
+        thumbnailUppy.reset();
+        setShowThumbnailDashboardModal(false);
+      });
+      thumbnailUppy.on('file-added', () => {
+        setFileInputIsBusy(true);
+      });
+    }
+  }, [thumbnailUppy]);
+
+  useEffect(() => {
+    const setNewThumbnailAPICallAndFormikChange = async () => {
+      try {
+        setFileInputIsBusy(true);
+        if (newThumbnailContent) {
+          let responseEvents = await getResourceContentEvent(newThumbnailContent.identifier);
+          let thumbnailReady = responseEvents.data.resource_events.find(
+            (event) => event.event === ResourceThumbnailUploaded
+          );
+          let numberOfTries = 0;
+          while (!thumbnailReady && numberOfTries < 4) {
+            await new Promise((r) => setTimeout(r, 1000));
+            responseEvents = await getResourceContentEvent(newThumbnailContent.identifier);
+            thumbnailReady = responseEvents.data.resource_events.find(
+              (event) => event.event === ResourceThumbnailUploaded
+            );
+            numberOfTries++;
+          }
+          await setContentAsDefaultThumbnail(values.identifier, newThumbnailContent.identifier);
+          let tobeDeletedIdentifier = '';
+          for (let i = 0; i < values.contents.length; i++) {
+            if (values.contents[i].identifier === newThumbnailContent.identifier) {
+              values.contents[i].features.dlr_thumbnail_default = 'true';
+            } else if (
+              values.contents[i].identifier !== newThumbnailContent.identifier &&
+              values.contents[i].features.dlr_thumbnail_default === 'true' &&
+              values.contents[i].features.dlr_content_master === 'false'
+            ) {
+              tobeDeletedIdentifier = values.contents[i].identifier;
+            } else {
+              values.contents[i].features.dlr_thumbnail_default = 'false';
```

Comment:
Denne biten med kode må finne ut om forrige content som var thumbnail var master content, hvis ikke slett kontenten. Hvis vi skiller ut master-content for seg selv ville denne biten vært litt kortere